src/khepri.erl

```diff
@@ -203,6 +203,19 @@
 -type child_list_length() :: non_neg_integer().
 %% Number of direct child nodes under a tree node.
 
+-type delete_reason() :: direct | keep_while.
```

Comment:
I find `direct` to be a non-obvious reason. How about

``` erl
-type delete_reason() :: explicit | condition_lapsed.
```

I don't know if `condition_lapsed` is clearer than `keep_while` but `explicit` seems more obvious than `direct` to me.

Another option I have is

``` erl
-type delete_reason() :: user_action | condition_lapsed.
```

So, a key can be deleted as a result of a user action or a lapsed condition.

Reply:
I like the `explicit` name.

`user_action` is unclear because it is a library, so there is no real users.

I prefer `keep_while` because it’s the name of the option to add the condition. Therefore the relation is explicit between the condition/option and the delete reason.

Reply:
I went for

```erl
-type delete_reason() :: explicit | keep_while.
```